src/searchEngines.js

```diff
@@ -0,0 +1,157 @@
+         * Array storing tuples of tabs name and their corresponding CSS selector
+         */
+        tabs: [
+            [
+                options.navigateSearchTab,
+                'a.q.qs:not([href*="&tbm="]):not([href*="maps.google."])'
+            ],
+            [options.navigateImagesTab, 'a.q.qs[href*="&tbm=isch"]'],
+            [options.navigateVideosTab, 'a.q.qs[href*="&tbm=vid"]'],
+            [options.navigateMapsTab, 'a.q.qs[href*="maps.google."]'],
+            [options.navigateNewsTab, 'a.q.qs[href*="&tbm=nws"]'],
+            [options.navigateShoppingTab, 'a.q.qs[href*="&tbm=shop"]'],
+            [options.navigateBooksTab, 'a.q.qs[href*="&tbm=bks"]'],
+            [options.navigateFlightsTab, 'a.q.qs[href*="&tbm=flm"]'],
+            [options.navigateFinancialTab, 'a.q.qs[href*="&tbm=fin"]'],
+            [options.navigatePreviousResultPage, '#pnprev'],
+            [options.navigateNextResultPage, '#pnnext']
+        ],
+        /**
+         * Get result links
+         * @returns a SearchResultCollection object
+         */
+        getSearchLinks () {
+            return new SearchResultCollection(
+                [
+                    [
+                    document.querySelectorAll('#search .r > a:first-of-type, #search .r g-link > a:first-of-type'),
+                    n => n.parentElement.parentElement
+                    ],
+                    [document.querySelectorAll('div.zjbNbe > a'), null],
+                    [document.querySelectorAll('div.eIuuYe a'), null], // shopping results
+                    [document.querySelectorAll('#pnprev, #pnnext'), null]
+                ],
+                [document.querySelectorAll('#search .kp-blk .r > a:first-of-type')]
+                );
+        },
+        /**
+         *  Filter the results based on date
+         * @param {*} period, filter identifier. Accpeted filter are :
+         *  'h' : get results from last hour
+         *  'd' : get result from last day
+         *  'w' : get results from last week
+         *  'm' : get result from last month
+         *  'y' : get result from last year
+         */
+        changeTools(period) {
+            // Save current period and sort.
+            const res = /&(tbs=qdr:.)(,sbd:.)?/.exec(location.href);
+            const currPeriod = (res && res[1]) || '';
+            const currSort = (res && res[2]) || '';
+            // Remove old period and sort.
+            const strippedHref = location.href.replace(/&tbs=qdr:.(,sbd:.)?/, '');
+            if (period) {
+                location.href =
+                strippedHref + (period === 'a' ? '' : '&tbs=qdr:' + period + currSort);
+            } else if (currPeriod) {
+                // Can't apply sort when not using period.
+                location.href =
+                strippedHref + '&' + currPeriod + (currSort ? '' : ',sbd:1');
+            }
+        }
+    },
+
+    // Startpage
+    {
+        urlPattern: /^(www|encrypted)\.startpage\./,
```

Comment:
Does `encrypted.startpage` exist?
I couldn't find it.

Reply:
It doesn't exist, just a bad copy/paste. Fixed in 731d1df